scripts/tests/twister/test_harness.py

```diff
@@ -34,6 +43,348 @@ def process_logs(harness, logs):
         harness.handle(line)
 
 
+TEST_DATA_11 = [('RunID: 12345', False, False, False, None, True),
+                ('PROJECT EXECUTION SUCCESSFUL', False, False, False, 'passed', False),
+                ('PROJECT EXECUTION SUCCESSFUL', True, False, False, 'failed', False),
+                ('PROJECT EXECUTION FAILED', False, False, False, 'failed', False),
+                ('ZEPHYR FATAL ERROR', False, True, False, None, False),
+                ('GCOV_COVERAGE_DUMP_START', None, None, True, None, False),
+                ('GCOV_COVERAGE_DUMP_END', None, None, False, None, False),]
+@pytest.mark.parametrize(
+    "line, fault, fail_on_fault, cap_cov, exp_stat, exp_id",
+    TEST_DATA_11,
+    ids=["match id", "passed passed", "passed failed", "failed failed", "fail on fault", "GCOV START", "GCOV END"]
+)
+def test_harness_process_test(line, fault, fail_on_fault, cap_cov, exp_stat, exp_id):
+    #Arrange
+    harness = Harness()
+    harness.run_id = 12345
+    harness.state = None
+    harness.fault = fault
+    harness.fail_on_fault = fail_on_fault
+
+    #Act
+    harness.process_test(line)
+
+    #Assert
+    assert harness.matched_run_id == exp_id
+    assert harness.state == exp_stat
+    assert harness.capture_coverage == cap_cov
+
+def test_robot_configure():
+    #Arrange
+    mock_platform = mock.Mock()
+    mock_platform.name = "mock_platform"
+    mock_testsuite = mock.Mock(id = 'id', testcases = [])
+    mock_testsuite.name = "mock_testsuite"
+    instance = TestInstance(testsuite=mock_testsuite, platform=mock_platform, outdir="")
+    instance.testsuite.harness_config = {
+        'robot_test_path': '/path/to/robot/test'
+    }
+    robot_harness = Robot()
```

Comment:
I would insert a newline between set-ups for different variables if they take up multiple lines. To illustrate my point, this section would be:

```python
    #Arrange
    mock_platform = mock.Mock()
    mock_platform.name = "mock_platform"
    
    mock_testsuite = mock.Mock(id = 'id', testcases = [])
    mock_testsuite.name = "mock_testsuite"
    
    instance = TestInstance(testsuite=mock_testsuite, platform=mock_platform, outdir="")
    instance.testsuite.harness_config = {
        'robot_test_path': '/path/to/robot/test'
    }
    
    robot_harness = Robot()
```

It should help out new readers recognise when a given variable is 'done' and thus help to understand the tests.